lib/archethic/bootstrap.ex

```diff
@@ -206,12 +208,93 @@ defmodule Archethic.Bootstrap do
       Logger.info("Synchronization finished")
     end
 
+    nodes = P2P.authorized_and_available_nodes()
+    # mint reward resync req to load gen addr from db
+    resync_network_chain([:oracle, :node_shared_secrets, :mint_rewards], nodes)
+
     Sync.publish_end_of_sync()
     SelfRepair.start_scheduler()
-
     :persistent_term.put(:archethic_up, :up)
   end
 
+  # when no node loaded in memtable
+  @spec resync_network_chain(any, maybe_improper_list) :: :ok
+  def resync_network_chain(_type_list, _nodes = []),
+    do: Logger.info("Enforced Resync of Network Txs: failure, No-Nodes")
+
+  def resync_network_chain(type_list, nodes) when is_list(nodes) do
+    # spawn process for each type of txn to sync
+    Task.Supervisor.async_stream_nolink(Archethic.TaskSupervisor, type_list, fn type ->
+      # acquire a(any) address of type txn, (for querying DB)
+      case TransactionChain.list_addresses_by_type(type) |> Enum.at(0) do
+        nil ->
+          # no txn avilable for that type txn
+          # can fetch that "type" network chain
+          Logger.info("Enforced Resync of Network Transactions: failure , Transaction_addr: NA ",
+            transaction_type: type
+          )
+
+          :ok
+
+        addr ->
+          # get last txn locally available
+          {last_addr, _dt} = TransactionChain.get_last_address(addr)
+
+          try do
+            # fetch txs remotely post last_addr
+            TransactionChain.stream_remotely(last_addr, nodes, last_addr)
```

Comment:
I think you can just use the function `Replication.validate_and_store_transaction_chain()` instead of `Replication.validate_and_store_transaction()` as it already fetch the missing part of the chain.
So to use it you can fetch remotely the last address of the chain and call the function

Reply:
No we want to avoid warnings that transaction already exist

Reply:
You just have to compare if the local last address is different than the remote last address. If so you will not write a transaction that already exists

Reply:
> You just have to compare if the local last address is different than the remote last address. If so you will not write a transaction that already exists

👍

Reply:
> > You just have to compare if the local last address is different than the remote last address. If so you will not write a transaction that already exists
> 
> +1


We are taking last addr and fetching with last addr as paging addr . What ever txs we will receive, will be post last addr, I added it if we ever had got those txns/